File: frontendv2/src/models/pedido.ts
interface ElementoDetalle {
  Id_elemento: number;
  Nombre: string;
  Descripcion: string;
}

interface PedidoDetalle {
  Id: number;
  Monto: number;
  Fecha_entrega: Date;
  Fecha_emision: Date;
  Id_cliente: number;
  Id_medio_pago: number;
  Id_florista: any;
  Id_delivery: number;
  Id_estado: number;
  Elementos: ElementoDetalle[];
}

interface ElementoPedido {
  Id_elemento: number;
  Nombre: string;
  Cantidad: number;
}

interface IPedido {
  Id: number;
  Monto: number;
  Fecha_emision: Date;
  Fecha_entrega: Date;
  Id_cliente: number;
  Correo: string;
  Telefono: string;
  Id_medio_pago: number;
  Id_florista: any;
  Id_delivery: number;
}

export type { PedidoDetalle, ElementoDetalle, IPedido, ElementoPedido };

File: backend/models/pedido.go
package models

import "time"

type ElementoPedido struct {
	Id_elemento int
	Nombre      string
	Cantidad    int
}

type Pedido struct {
	Id            int
	Monto         float64
	Fecha_emision time.Time
	Fecha_entrega time.Time
	Id_cliente    int
	Correo        string
	Telefono      string
	Id_medio_pago int
	Id_florista   *int
	Id_delivery   *int
	Id_estado     int
}

type ElementoDetalle struct {
	Id_elemento int
	Nombre      string
	Descripcion string
}

type PedidoDetalle struct {
	Id            int
	Monto         float64
	Fecha_entrega time.Time
	Fecha_emision time.Time
	Id_cliente    int
	Id_medio_pago int
	Id_florista   *int
	Id_delivery   *int
	Id_estado     int
	Elementos     []ElementoDetalle
}
